src/services/arweave/WarpDataProvider.ts

```diff
@@ -79,4 +96,97 @@ export class WarpDataProvider implements SmartweaveDataProvider {
     const state = await this.getContractState(id);
     return state?.balances[wallet.toString()] ?? 0;
   }
+  /**
+   *
+   * @param srcCodeTransactionId - ArweaveTransactionId of the source code used to deploy contract
+   * @param initialState - typeof ANTContractState or ArweaveTransactionID - depending on the stateType chosen, either the json initial state of the contract to deploy, or the on-chain states TXID
+   * @param stateType - type of 'TXID' | 'TAG' | 'DATA' - TXID indicates use of a prexisting state on-chain, TAG and DATA use json states locally. DATA puts the state in the data key of the transaction, TAG puts the state in the tag.
+   * @returns
+   */
+  async deployContract({
+    srcCodeTransactionId,
+    initialState,
+    stateType,
+    tags = [],
+  }: {
+    srcCodeTransactionId: ArweaveTransactionID;
+    initialState: ANTContractState | ArweaveTransactionID;
+    stateType: 'TXID' | 'TAG' | 'DATA';
+    tags: TransactionTag[];
+  }): Promise<string | undefined> {
+    const stateSize = new Blob([JSON.stringify(initialState)]).size;
+    const tagSize = new Blob([JSON.stringify(tags)]).size;
+
+    try {
+      if (!initialState) {
+        throw new Error('Must have an initial state to deploy a contract');
+      }
+      if (
+        isArweaveTransactionID(initialState.toString()) &&
+        stateType !== 'TXID'
+      ) {
+        throw new Error(
+          'Mismatching initial state and state type. TXIDs are not valid contract states. If the intent was to use an on chain state, set the stateType parameter to TXID',
+        );
+      }
+      if (stateSize > SMARTWEAVE_MAX_TAG_SPACE && stateType === 'TAG') {
+        throw new Error(
+          `state too large for tag space, must be under ${SMARTWEAVE_MAX_TAG_SPACE} bytes.`,
+        );
+      }
+      if (tagSize > SMARTWEAVE_MAX_TAG_SPACE) {
+        throw new Error(
+          `tags too large for tag space, must be under ${SMARTWEAVE_MAX_TAG_SPACE} bytes.`,
+        );
+      }
+      if (
+        stateSize + tagSize > SMARTWEAVE_MAX_TAG_SPACE &&
+        stateType !== 'TXID'
+      ) {
+        throw new Error(
+          `Combined size of tags (${tagSize} bytes) and state (${stateSize} bytes) too large for tag space, must be under ${SMARTWEAVE_MAX_TAG_SPACE} bytes. Reduce the tags or use a different state deployment method.`,
+        );
+      }
+      const deploymentPayload: {
+        wallet: ArWallet;
+        initState: string;
+        srcTxId: string;
+        tags: TransactionTag[];
+      } = {
+        wallet: 'use_wallet',
+        initState: '',
+        srcTxId: srcCodeTransactionId.toString(),
+        tags: tags,
+      };
```

Comment:
nit: prefer to use `Map` when we are creating an object and dynamically updating it's values in these classes